sdk/eventgrid/eventgrid/src/eventGridClient.ts

```diff
@@ -140,10 +163,25 @@ export class EventGridPublisherClient<T extends InputSchema> {
           );
         }
         case "CloudEvent": {
+          // The underlying REST API expects a header named `aeg-channel-name`, and so the generated client
+          // expects that options bag has a property called `aegChannelName`, where as we expose it with the
+          // friendlier name "channelName". Fix up the impedence mismatch here
+          const { channelName, ...sendOptions } = {
+            ...(updatedOptions as CloudEventSendOptions),
```

Comment:
does the spread operator do anything useful here? Spreading and then destructuring seems pretty much the same as directly destructuring in terms of side effects?

Reply:
No, you're right, this is all very goofy, changed this to:

```typescript
          const { channelName, ...sendOptions } = updatedOptions as CloudEventSendOptions;
```